Library/Homebrew/extend/ENV/super.rb

```diff
@@ -74,7 +74,7 @@ def setup_build_environment(formula: nil, cc: nil, build_bottle: false, bottle_a
     self["CMAKE_INCLUDE_PATH"] = determine_cmake_include_path
     self["CMAKE_LIBRARY_PATH"] = determine_cmake_library_path
     self["ACLOCAL_PATH"] = determine_aclocal_path
-    self["M4"] = DevelopmentTools.locate("m4") if deps.any? { |d| d.name == "autoconf" }
+    self["M4"] = "#{HOMEBREW_PREFIX}/opt/m4/bin/m4" if deps.any? { |d| d.name == "autoconf" }
```

Comment:
As indicated in https://github.com/Homebrew/homebrew-core/pull/73797#issuecomment-805691769 I think it's right to remove this now.

However, I think the line should be removed, not hardcoding `HOMEBREW_PREFIX`. We already have all the mechanism in place to put our m4 ahead in the PATH when it's a recursive dependency, don't we?

Reply:
This seems to be used on Linux too, and returns a path to brewed `m4` there. Not sure if it's needed there, that's why I didn't remove it.

Reply:
@Homebrew/linux, anyone opposed to just removing the `M4` variable in superenv?

Reply:
What would it change for us? I am trying to find out what happened with this line in the past and why it impacts Linux.

Reply:
I guess we can remove it and find out. (It should not do any harm.)

Reply:
It seems to set `M4` to the path of brewed `m4` here:

https://github.com/Homebrew/brew/blob/ed87211daa83982a6136e14d181a1550b46a0f17/Library/Homebrew/extend/os/linux/development_tools.rb#L8-L16

Reply:
Ok, after some testing (thanks @iMichka), this seems to be good on Linux. Let's remove it.